internal/femto/femto.go

```diff
@@ -30,15 +30,42 @@ func OnGet[T any](f *Femto, pattern string, handle GetFunc[T]) {
 	})
 }
 
+// correctMethod returns wether the request had the given method.
+//
+// If not, it will return a suitable reponce: either No Content for an OPTIONS request,
+// or
+func correctMethod(method string, req *http.Request, w http.ResponseWriter, log *slog.Logger) bool {
+	switch req.Method {
+	case method:
+		return true
+	case http.MethodOptions:
+
+		// We don't set Access-Control-Allow-Origin, as that'd done globally
+		// so it can be enabled for dev, but not prod.
+
+		// https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+
+		w.Header().Set("Allow", method)
+		w.Header().Set("Access-Control-Allow-Methods", method)
+		w.Header().Set("Access-Control-Allow-Headers", "Authorization, Content-Type")
```

Comment:
`Content-Type` is a [CORS-safelisted Request Header](https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_request_header) that's always allowed, so doesn't have to be listed in `Access-Control-Allow-Headers` unless we want to bypass the additional restrictions (ie. we want to allow non-mime types here)

Reply:
Hmmm, for some reason this mattered though:

![image](https://github.com/gpuctl/gpuctl/assets/28781354/fae2bc52-f9ef-47d1-8f2e-152967283fd6)

![image](https://github.com/gpuctl/gpuctl/assets/28781354/7252c174-b993-4801-a99b-98753238a873)

And the `Authorization` header wasn't set in the request we were fighting with. I just added it because I figured it'd come up.